docs/content/guides/developer/app-examples/trustless-swap.mdx

```diff
@@ -996,49 +1068,151 @@ The UI design consists of three parts:
+The first step is to set up the client app. Run the following command to scaffold a new app.
 
-:::
+```bash
+pnpm create @mysten/dapp --template react-client-dapp
+```
 
-### Set up providers {#set-up-providers}
+or
 
-Set up and configure several providers at the root of your React.js tree to ensure different libraries including `dApp Kit`, `@tanstack/react-query`, `react-router-dom` work as expected.
+```bash
+yarn create @mysten/dapp --template react-client-dapp
+```
 
-```ts title='src/main.tsx'
-import { createNetworkConfig, SuiClientProvider, WalletProvider } from '@mysten/dapp-kit';
-import { getFullnodeUrl } from '@mysten/sui/client';
-import { Theme } from '@radix-ui/themes';
-import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
-import { RouterProvider } from 'react-router-dom';
+### Setting up import aliases
 
-import { router } from '@/routes/index.tsx';
+First, set up import aliases to make the code more readable and maintainable. This allows you to import files using `@/` instead of relative paths.
 
-const queryClient = new QueryClient();
+Replace the content of `tsconfig.json` with the following:
 
-const { networkConfig } = createNetworkConfig({
-	localnet: { url: getFullnodeUrl('localnet') },
-	devnet: { url: getFullnodeUrl('devnet') },
-	testnet: { url: getFullnodeUrl('testnet') },
-	mainnet: { url: getFullnodeUrl('mainnet') },
-});
+{@inject: examples/trading/frontend/tsconfig.json}
+
+Replace the content of `vite.config.ts` with the following:
 
-ReactDOM.createRoot(document.getElementById('root')!).render(
-	<React.StrictMode>
-		<Theme appearance="light">
-			<QueryClientProvider client={queryClient}>
-				<SuiClientProvider networks={networkConfig} defaultNetwork="testnet">
-					<WalletProvider autoConnect>
-						<RouterProvider router={router} />
-					</WalletProvider>
-				</SuiClientProvider>
-			</QueryClientProvider>
-		</Theme>
-	</React.StrictMode>,
-);
+{@inject: examples/trading/frontend/vite.config.ts}
+
+### Adding Tailwind CSS
+
+This app uses Tailwind CSS for styling. Run the following command to add Tailwind CSS and its dependencies:
+
+```bash
+pnpm add tailwindcss@latest postcss@latest autoprefixer@latest
+```
+
+or
+
+```bash
+yarn add tailwindcss@latest postcss@latest autoprefixer@latest
+```
+
+Next, generate the Tailwind CSS configuration file by running:
+
+```bash
+npx tailwindcss init -p
 ```
 
+Replace the content of `tailwind.config.js` with the following:
+
+{@inject: examples/trading/frontend/tailwind.config.js}
+
+Add the `src/styles/` directory and add `base.css`: 
+
+{@inject: examples/trading/frontend/src/styles/base.css}
+
+### Connecting your deployed package
+
+First, deploy your package via the [scripts in the api directory](#deployment).
+
+Then, create a `src/constants.ts` file and fill it with the following:
+
+{@inject: examples/trading/frontend/src/constants.ts}
+
+### Add helper functions and UI components
```

Comment:
@ronny-mysten can we find a way to collapse these sections down? It's a lot of code to scroll through without necessarily having to understand it. Initially I was going to suggest maybe just linking to these files and asking folks to copy them from Github but that extra level of friction doesn't seem ideal.

I was imagining a component that, when expanded, showed the code block, but when collapsed, showed only the file name, and maybe a link to it in GitHub, and in either case offered a button to copy the contents. That way, by default, we don't take up a lot of vertical space for code that people need, but that we don't really want to explain too deeply.